src/serialization/deserialize-assets.js

```diff
@@ -106,13 +110,44 @@ const deserializeCostume = function (costume, runtime, zip, assetFileName) {
         return Promise.resolve(null);
     }
 
-    return costumeFile.async('uint8array').then(data => storage.createAsset(
-        assetType,
-        // TODO eventually we want to map non-png's to their actual file types?
-        costumeFormat,
-        data,
-        assetId
-    ));
+    // textLayerMD5 exists if there is a text layer, which is a png of text from Scratch 1.4
+    // that was opened in Scratch 2.0. In this case, set costume.textLayerAsset.
+    let textLayerFilePromise;
+    if (costume.textLayerMD5) {
+        textLayerFileName = textLayerFileName ? textLayerFileName :
+            `${costume.textLayerMD5}.png`;
```

Comment:
Does textLayer info behave the same way that baseLayer info does? E.g. in sb2 files, there's a baseLayerMD5 which is the asset ID and there's the actual file name which is something like 0.png, does the textLayer also have a similar concept or is its file name assetID + extension?

Mostly I'm wondering if this `textLayerFileName` param is necessary.

Reply:
I think textLayerFileName works exactly the same way as assetFileName. I'm not sure what you mean about it not being necessary. Is there somewhere else I should be getting it?

Reply:
The reason `assetFileName` is provided as an argument to this function and the reason that 
https://github.com/LLK/scratch-vm/pull/1737/files#diff-7fd991d2b1bb7163e8a10acfcad7c3e3R68
exists is because sb2s and sb3s have different file name structures. The call to `deserializeCostume` from the sb3 deserialization code does not provide the `assetFileName` argument, so the `or` case in that line is for the `sb3` case.

Since we only expect the text layer special case to be part of the sb2 deserialization process, we can probably just have the sb2 import code add the filename to the costume as a temporary property that we look for here and then delete it when this function successfully deserializes the costume. 

The downside of that is that we could refactor the code in deserializeCostume later down the line and forget to delete the properties, so I'm also okay to leave the argument as is, but I think this line (e.g. ``textLayerFileName = textLayerFileName ? textLayerFileName : `${costume.textLayerMD5}.png`;``) should be removed.